test/extended/networking/egressip.go

```diff
@@ -0,0 +1,654 @@
+	// b) We make sure that the request leaves from one of the EgressIPs
+	// Therefore, we can send our request against any destination.
+	// If we tests against any TCP/HTTP endpoint, we can additionally
+	// c) Prove that we established a bidirectional stream.
+	framework.Logf("Launching a new prober pod and sending requests with a unique search string")
+	searchString, err := sendProbesToHostPort(oc, externalNamespace, probePodName, routeName, targetProtocol, targetHost, targetPort, iterations)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	// Tcpdump runs with -l for line buffered and the container runs with a TTY. This sleep here should not be necessary,
+	// but let's have it as a safety measure and to avoid that the test itself causes failures.
+	framework.Logf("Sleeping for 10 seconds to give container logs and tcpdump a little bit of time to refresh")
+	time.Sleep(10 * time.Second)
+
+	framework.Logf("Making sure that %d requests with search string %s and EgressIPs %v were seen", expectedHits, searchString, egressIPSet)
+	numberFound := 0
+	found, err := scanPacketSnifferDaemonSetPodLogs(oc, packetSnifferDaemonSet, targetProtocol, searchString)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	framework.Logf("Found map is: %v", found)
+	for egressIP := range egressIPSet {
+		if n, ok := found[egressIP]; ok {
+			framework.Logf("Found EgressIP %s for string %s %d times", egressIP, searchString, n)
+			numberFound += n
+		}
+	}
+	o.Expect(func() bool { return numberFound == expectedHits }()).To(o.BeTrue())
+}
+
+// applyEgressIPObject is a wrapper that applies the EgressIP object in file <egressIPYamlPath> with name <egressIPNamespace>
+// The propagation from a created EgressIP object to CloudPrivateIPConfig can take quite some time on Azure, hence also add a
+// check that waits for the CloudPrivateIPConfigs to be created.
+func applyEgressIPObject(oc *exutil.CLI, egressIPYamlPath, egressIPNamespace string, egressIPSet map[string]struct{}, timeout int) {
+	framework.Logf("Applying the EgressIP object %s", egressIPNamespace)
+	_, err := oc.AsAdmin().Run("apply").Args("-f", egressIPYamlPath).Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	// TODO: make this actually probe for both the EgressIP object *and* for the cloudprivateipconfigs
+	// created by that object instead of using a timer.
+	framework.Logf(fmt.Sprintf("Waiting for CloudPrivateIPConfig creation for a maximum of %d seconds", timeout))
+	var exists bool
+	o.Eventually(func() bool {
+		for eip := range egressIPSet {
+			exists, err = cloudPrivateIpConfigExists(oc, eip)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if !exists {
+				framework.Logf("CloudPrivateIPConfig for %s not found.", eip)
+				return false
+			}
+		}
+		framework.Logf("CloudPrivateIPConfigs for %v found.", egressIPSet)
+		return true
+	}, time.Duration(timeout)*time.Second, 5*time.Second).Should(o.BeTrue())
+
+	// There's some additional delay in play here. If we probe right after the above check,
```

Comment:
don't like having a sleep at the end of a function. would it make sense to wrap whatever the "probe" is in some
Eventually() that checks once a second for 15s? or something along those lines.

Reply:
Done (in both occurrences where I used a sleep instead of a retry mechanism)